src/test/java/uk/ac/ic/wlgitbridge/bridge/swap/job/SwapJobImplTest.java

```diff
@@ -120,11 +121,42 @@ public void correctProjGetSwappedWhenOverHighWatermark(
         while (swapJob.swaps.get() < 1);
         assertEquals(1, dbStore.getNumUnswappedProjects());
         assertEquals("proj1", dbStore.getOldestUnswappedProject());
+        assertEquals("bzip2", dbStore.getSwapCompression("proj2"));
         swapJob.restore("proj2");
+        assertEquals(null, dbStore.getSwapCompression("proj2"));
         int numSwaps = swapJob.swaps.get();
         while (swapJob.swaps.get() <= numSwaps);
         assertEquals(1, dbStore.getNumUnswappedProjects());
         assertEquals("proj2", dbStore.getOldestUnswappedProject());
     }
 
-}
\ No newline at end of file
+    @Test
+    public void swapCompressionGzip() throws IOException {
+        swapJob = new SwapJobImpl(
+                1,
+                15000,
+                30000,
+                Duration.ofMillis(100),
+                SwapJob.CompressionMethod.Gzip,
+                lock,
+                repoStore,
+                dbStore,
+                swapStore
+        );
+        swapJob.lowWatermarkBytes = 16384;
+        assertEquals(2, dbStore.getNumUnswappedProjects());
+        assertEquals("proj2", dbStore.getOldestUnswappedProject());
+        swapJob.start();
+        while (swapJob.swaps.get() < 1);
```

Comment:
Just curious. I see it's done like that in other tests. Is that loop waiting for the swap to have happened in another thread? Could it be made to wait a bit between iterations?

Reply:
Seems to be the case, yes. I'll see if we can put a little `Thread.sleep()` in there.